core/src/main/java/org/jdbi/v3/core/array/SqlArrayType.java

```diff
@@ -35,4 +37,14 @@
      * @return the converted element
      */
     Object convertArrayElement(T element);
+
+    /**
+     * Create a SqlArrayType from the given type and convert function.
+     *
+     * @param typeName the vendor sql type to use
+     * @conversion convert the array element to the jdbc representation
+     */
+    static <T> SqlArrayType<T> of(String typeName, Function<? super T, ?> conversion) {
```

Comment:
Not sure about the `? super T` wildcard. e.g. if you call with a lambda:

```java
SqlArrayType.<MyType>of(
    "mytype",
    value -> /* can't call MyType methods because the lower bound is Object */)
```

Reply:
Ah, bummer.  I was thinking that accepting any supertype is fine, for example a `Function<Object, Object>` is just fine for binding a `String`.  But if it breaks lambdas, that's not ok...